dvc/scm/git/__init__.py

```diff
@@ -333,8 +334,8 @@ def _get_diff_trees(self, a_ref, b_ref):
             # /en/2.1.11/reference.html#git.objects.base.Object.__str__
             commits.append(a_commit)
             commits.append(b_commit)
-            trees[DIFF_A_TREE] = self.get_tree(commits[0])
-            trees[DIFF_B_TREE] = self.get_tree(commits[1])
+            trees[DIFF_A_TREE] = CleanTree(self.get_tree(commits[0]))
+            trees[DIFF_B_TREE] = CleanTree(self.get_tree(commits[1]))
```

Comment:
SCM code should not know anything about clean trees.

Reply:
If diff will not be `dvcignore`-aware, we will get different tree for a branch when using `diff`, and when traversing it using `dvcignore`-wrapped GitTree, I don't see a way to make GitTree completely unaware of `dvcignore`

Reply:
Not really, we can wrap this into `CleanTree` later in a repo method.